### PR TITLE
Drop aws kubelet credential provider and cleanup aws storage e2e tests

### PR DESCRIPTION
**Pull request by dims**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
To get rid of dependency to aws-sdk library we need additional clean up. Also note:
- Since `ecr-credential-provider` has been available for a while and the [KEP-2133](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2133-kubelet-credential-providers) for external kubelet credential providers is already GA, folks can use that. 
- Also since we already removed the in-tree cloud provider for AWS, it only makes sense to clean the built-in kubelet credential provider as well.
- The storage e2e tests had some references as well which do not make sense now since we removed the AWS EBS support already.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Removed AWS kubelet credential provider. Please use the external kubelet credential provider binary named `ecr-credential-provider` instead.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by tzneal** [comment]
I've got a PR at https://github.com/kubernetes/kubernetes/pull/116236 that uses the AWS modules being removed here to implement a remote E2E node test runner like the existing GCE one.  I later found some discussion on https://github.com/kubernetes/kubernetes/issues/115262 where @bobbypage said to avoid using the remote runner since it's not what the actual CI jobs do.

Happy to close my PR, but should the remote runner support be removed entirely?

File: test/e2e/storage/volume_provisioning.go [hunk 1]
```diff
@@ -191,99 +130,6 @@ var _ = utils.SIGDescribe("Dynamic Provisioning", func() {
...
-						"zone": getRandomClusterZone(ctx, c),
-					},
-					ClaimSize:    "1.5Gi",
-					ExpectedSize: "2Gi",
-					PvCheck: func(ctx context.Context, claim *v1.PersistentVolumeClaim) {
-						volume := testsuites.PVWriteReadSingleNodeCheck(ctx, c, f.Timeouts, claim, e2epod.NodeSelection{})
-						gomega.Expect(volume).NotTo(gomega.BeNil(), "get bound PV")
-
-						err := checkAWSEBS(volume, "gp2", false)
-						framework.ExpectNoError(err, "checkAWSEBS gp2")
```
Comment (jsafrane) [inline test/e2e/storage/volume_provisioning.go:211]: I think only `checkAWSEBS()` function and these two lines should be removed - we still want to test that `type: gp2` in a StorageClass provisions a valid RWO volume. Without AWS SDK we can't check that it's really gp2, but IMO it's still better than nothing. The same applies to all AWS tests below.

**Comment by liggitt** [comment]
> should the remote runner support be removed entirely?

if it requires linking in cloud provider libraries, yes

File: test/e2e/storage/volume_provisioning.go (hunk 1 above)
Comment (dims) [reply test/e2e/storage/volume_provisioning.go:211]: @jsafrane ack, will do this.

**Comment by dims** [comment]
> regardless, I'm super happy to see the vendored packages drop out

+100 - am happy with 260k less lines of code in our repo!

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

from node

**Comment by xing-yang** [comment]
lgtm

**Comment by dims** [comment]
> lgtm

thanks @xing-yang 

/hold cancel

**Review by torredil — APPROVED** [review]
/lgtm

**Comment by dims** [comment]
Applied the suggestion. thanks @sftim

**Comment by buckleyGI** [comment]
Thanks @dims We haven't gotten that feedback at this point from support. We had a chat with them some hours ago so I suppose we will get guidance.

**Pull request merged**